aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -4,6 +4,71 @@
 
 #include <TH/generic/THTensorApply.hpp>
 
+// Finds non-zero elements of a tensor and returns their subscripts
+void THTensor_(nonzero)(THLongTensor *subscript, THTensor *tensor)
+{
+  ptrdiff_t numel = 0;
+  int64_t *subscript_data;
+  int64_t i = 0;
+#ifdef TH_REAL_IS_HALF
+#define IS_NONZERO(val) ((val.x & 0x7fff) != 0)
+#else
+#define IS_NONZERO(val) ((val)!=0)
+#endif
+
+  /* First Pass to determine size of subscripts */
+  TH_TENSOR_APPLY(scalar_t, tensor,
+                  if IS_NONZERO(*tensor_data) {
+                    ++numel;
+                  });
+#ifdef DEBUG
+  THAssert(numel <= LONG_MAX);
```

Comment:
did you change this at all?

Reply:
no, i didnt. i just moved it outside of 
#if !defined(TH_REAL_IS_BOOL) /* non bool only part */ as its needed for the indexing.